app/forms/degree_form.rb

```diff
@@ -18,23 +18,31 @@ class DegreeForm
     country
   ].freeze
 
-  attr_accessor(*FIELDS, :degrees_form, :degree)
+  AUTOCOMPLETE_FIELDS = %i[
+    uk_degree_raw
+    subject_raw
+    institution_raw
+  ].freeze
+
+  attr_accessor(*FIELDS, *AUTOCOMPLETE_FIELDS, :degrees_form, :degree)
 
+  validates :uk_degree, :subject, :institution, autocomplete: true, allow_nil: true
   validate :validate_with_degree_model
 
-  delegate :uk?, :non_uk?, :non_uk_degree_non_naric?,
+  delegate :uk?, :non_uk?, :non_uk_degree_non_naric?, :persisted?,
            to: :degree
 
-  def initialize(degrees_form:, degree:)
+  def initialize(degrees_form:, degree:, autocomplete_params: {})
     @degrees_form = degrees_form
     @degree = degree
     self.attributes = degree.attributes
       .symbolize_keys
       .slice(*FIELDS)
+    assign_attributes(autocomplete_params)
   end
 
-  def persisted?
-    slug.present?
+  def self.model_name
```

Comment:
For my own understand - why does this particular form need the model name set?

Reply:
The form partial used for degree details is shared across the edit and create actions, so we let rails figure out the right url and http method to use rather than passing it in explicitly as we do in many other forms, eg: https://github.com/DFE-Digital/register-trainee-teachers/pull/768/files#diff-ab024b8a7a437f43221059fb76deb93f5c34b0bf6d5ede44efe4b2876eef5d80R1

If we don't set the model name here rails tries to generate a url like `trainee_degree_form_path`, which doesn't exist.

Reply:
Understood!